crates/functional-test-utils/src/drt.rs

```diff
@@ -0,0 +1,280 @@
+    };
+    use strata_common::logging;
+    use tracing::debug;
+
+    use super::*;
+
+    const SECRET_KEY: &str = "0000000000000000000000000000000000000000000000000000000000000001";
+    const EL_ADDRESS: &str = "deedf001900dca3ebeefdeadf001900dca3ebeef";
+    const AGGREGATED_OPERATOR_PUBLIC_KEY: &str =
+        "02b4634c515a62e47b3f3eb62b8a6f6320fdb2baed5f2e6657f472b0f2a33221";
+
+    /// Get the authentication credentials for a given `bitcoind` instance.
+    fn get_auth(bitcoind: &BitcoinD) -> (String, String) {
+        let params = &bitcoind.params;
+        let cookie_values = params.get_cookie_values().unwrap().unwrap();
+        (cookie_values.user, cookie_values.password)
+    }
+
+    /// Mine a number of blocks of a given size `count`, which may be specified to a given coinbase
+    /// `address`.
+    fn mine_blocks(
+        bitcoind: &BitcoinD,
+        count: usize,
+        address: Option<Address>,
+    ) -> anyhow::Result<Vec<BlockHash>> {
+        let coinbase_address = match address {
+            Some(address) => address,
+            None => bitcoind
+                .client
+                .get_new_address(None, None)?
+                .assume_checked(),
+        };
+        let block_hashes = bitcoind
+            .client
+            .generate_to_address(count as _, &coinbase_address)?;
+        Ok(block_hashes)
+    }
+
+    #[tokio::test()]
+    async fn drt_mempool_accept() {
+        // FIXME: remove the debug! statements.
+        logging::init();
+
+        let bitcoind = BitcoinD::from_downloaded().unwrap();
+        let url = bitcoind.rpc_url();
+        let (user, password) = get_auth(&bitcoind);
+        let client = BitcoinClient::new(url, user, password).unwrap();
+
+        let secret_key = SECRET_KEY.parse::<SecretKey>().unwrap();
+        let public_key = even_x_only_pubkey(&secret_key);
+        let p2tr_address = p2tr_address(&public_key);
+
+        // Mine and get the last UTXO which should have 50 BTC.
+        let block_hash = mine_blocks(&bitcoind, 101, Some(p2tr_address))
+            .unwrap()
+            .pop()
+            .unwrap();
+        debug!(%block_hash, "Mined blocks");
+        let block = client.get_block(&block_hash).await.unwrap();
+        debug!(block_txdata = ?block.txdata, "Block TX data");
+        let block_tx = block.txdata[0].clone();
+        debug!(?block_tx, "Block transaction");
+        let txid = block_tx.compute_txid();
+        let outpoint = OutPoint { txid, vout: 0 };
+        debug!(%outpoint, "Outpoint");
+
+        let signed_tx = deposit_request_transaction_inner(
+            SECRET_KEY.to_string(),
+            EL_ADDRESS.to_string(),
+            AGGREGATED_OPERATOR_PUBLIC_KEY.to_string(),
+            outpoint.to_string(),
+        )
+        .unwrap();
+
+        // Wait for coinbase to mature
+        // let _ = mine_blocks(&bitcoind, 6, None).unwrap();
+
+        let txid = client.send_raw_transaction(&signed_tx).await.unwrap();
```

Comment:
Ok this explodes with:

```console
Server(-26, "bad-txns-premature-spend-of-coinbase, tried to spend coinbase at depth 1")
```

At least the signing is working and the proper DRT tx building as well.
Gonna debug further